examples/example-gateway/gateway.json

```diff
@@ -13,5 +13,12 @@
 
 	"annotationPrefix": "zanzibar",
 	"traceKey": "x-trace-id",
-	"genMock": true
+	"genMock": true,
+
+	"defaultReqHeaderMap": {
+		"x-trace-id": "x-trace-id"
```

Comment:
Should we have this config as a map? To me, this is saying "**transform** endpoint header `x-trace-id` to client header `x-trace-id` for every endpoint". I think a global setting of forwarding is more proper:

```
"defaultReqHeadersToForward": [
    "x-trace-id"
]
```
This says "forward endpoint header `x-trace-id` to client header for each endpoint".

I think it is important to express the config semantic clearly. It's fine that header forwarding is achieved/implemented via transformation with the same header key, but the config semantic should be clear that it's forwarding.

Reply:
I do agree that a global setting of forwarding is more proper. I just thought that I might as well try to follow the style of the endpoint configs, which have a `reqHeaderMap` and `resHeaderMap`, so it'd be consistent to have a `default[Req/Res]HeaderMap` that behaves the same way. What do you think?

Reply:
@cl1337